src/librbd/migration/NativeFormat.cc

```diff
@@ -151,30 +200,40 @@ void NativeFormat<I>::open(Context* on_finish) {
 
   if (snap_id_val.type() != json_spirit::null_type &&
       m_snap_id == CEPH_NOSNAP) {
-    lderr(cct) << "invalid snap id" << dendl;
+    lderr(m_cct) << "invalid snap id" << dendl;
     on_finish->complete(-EINVAL);
     return;
   }
 
   // snapshot is required for import to keep source read-only
   if (m_import_only && m_snap_name.empty() && m_snap_id == CEPH_NOSNAP) {
-    lderr(cct) << "snapshot required for import" << dendl;
+    lderr(m_cct) << "snapshot required for import" << dendl;
     on_finish->complete(-EINVAL);
     return;
   }
 
-  // TODO add support for external clusters
   librados::IoCtx io_ctx;
-  int r = util::create_ioctx(m_image_ctx->md_ctx, "source image",
-                             m_pool_id, m_pool_namespace, &io_ctx);
+  int r = rados_ptr->ioctx_create2(m_pool_id, io_ctx);
   if (r < 0) {
+    if (r == -ENOENT) {
+      lderr(m_cct) << "source image pool " << m_pool_id << " does not exist"
+                   << dendl;
+    } else {
+      lderr(m_cct) << "error accessing source image pool " << m_pool_id
+                   << dendl;
```

Comment:
It would be nice to dump the actual error here.  Let's replace this entire `if` with `lderr(m_cct) << "error accessing source image pool " << m_pool_id << ": " << cpp_strerror(r) << dendl;`.